packages/core/core/src/public/utils.js

```diff
@@ -0,0 +1,17 @@
+// @flow strict-local
+
+import assert from 'assert';
+
+export function getPublicId(id: string): string {
+  assert(
+    id.match(/^[0-9a-f]{32}$/),
+    `id ${id} must be a 32-character hexadecimal string`,
+  );
+
+  let buf = Buffer.alloc(8);
+  // Add the integer values to the buffer
+  // https://stackoverflow.com/questions/8044543/how-can-i-store-an-integer-in-a-nodejs-buffer/53550757#53550757
+  buf.writeUInt32BE(parseInt(id.slice(0, 8), 16), 0);
+  buf.writeUInt32BE(parseInt(id.slice(8, 16), 16), 4);
+  return buf.toString('base64').slice(0, 6);
```

Comment:
We could also use the url-friendly base64url [0], and the namer could even use that in the bundle's filename as well. Or maybe base62 — at least in the bundle case — to avoid special characters all together.

[0] https://tools.ietf.org/html/rfc4648#page-7

Reply:
I don't think we should shrink the id at all, it will probably reintroduce the issues we had in parcel 1, https://github.com/parcel-bundler/parcel/issues/2180 and https://github.com/parcel-bundler/parcel/pull/2999

We could encode it with base62 or something to not lose any data as we don't have to remove any non-alphanumeric characters

Reply:
I like the idea of using base62 here combined with using @devongovett's suggestion of only ever exposing the public id. base64 won't be safe to use in a variable name, but as long as there's a prefix (variable names can't begin with a digit), base62 would work in that case as well.